templates/cicd/pipeline_cfn.yml

```
@@ -179,6 +179,22 @@ Resources:
               - arn:aws:iam::{{$stage.AccountID}}:role/{{$.AppName}}-{{$stage.Name}}-EnvManagerRole{{end}}
       Roles:
         - !Ref PipelineRole
+  {{$length := len .Stages}}{{if gt $length 0}}{{range $index, $stage := .Stages}}
+  BuildTestCommands{{$index}}:
+    Type: AWS::CodeBuild::Project
+    Properties:
+      Name: BuildTestCommands{{$index}}
```

I think instead of using indices to coordinate between the pipeline stage and these codebuild project, we can use the stage name and that should be a less bug-prone route.

For example, not every stage need to have `test_commands` in them, so finding the right index becomes difficult I think

```suggestion
  BuildTestCommands{{$stage.Name}}:
    Type: AWS::CodeBuild::Project
    Properties:
      Name: BuildTestCommands-{{$stage.Name}}
```

---

Good idea. Done!